packages/core-data/src/actions.js

```diff
@@ -57,6 +57,20 @@ export function receiveCurrentUser( currentUser ) {
 	};
 }
 
+/**
+ * Returns an action object used to lock the editor.
+ *
+ * @param {Object} currentUserSession Meta information about the login session for current user.
+ *
+ * @return {Object} Action object.
+ */
+export function __experimentalUpdateCurrentUserSession( currentUserSession ) {
```

Comment:
This looks like local state. core-data was not really mean to store things like that. So can you explain more the reasoning here and what this is about?

Reply:
I think this state could be very easily moved to the `editor` store, to the `state.postLock` object. That object already has plenty of fields (`isLocked`, `isTakeover`, `user`, ...), so why not add another `isUserActive` field there?

All other aspects of this user session check should be integrated into the post lock state, too. Why add another hook to `heartbeat.tick`, if we can just expand the existing `receivePostLock` function?

Reply:
Moving this to the `postLock` object sounds confusing. Post lock state is coming from an entirely different API response and it's easy to accidentally override `postLock.isUserActive` (or vice-versa) in the process.

Reply:
Then it could be renamed to `state.heartbeat`, with subfields for post lock, auth check and possibly other things. They are all requested together, and also used together.